### PR TITLE
Add Jenkins stage for Docker build and push

### PR DESCRIPTION
**Pull request by anoopwebs**
**What this PR does / why we need it**:

Docker builds and push are not part of Jenkins build now so it's unclear when a change gets to build and pushed to docker registry. 

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:
The recent change in the repo https://github.com/caicloud/event_exporter/pull/9 hasn't got into docker registry yet, so thinking to handle it in Jenkins build.

Fixes #

Reference to #

**Special notes for your reviewer**:
 * Please add proper docker registry secret in https://github.com/caicloud/event_exporter/compare/master...anoopwebs:patch-1?expand=1#diff-58231b16fdee45a03a4ee3cf94a9f2c3R43

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -30,12 +30,28 @@ podTemplate(
             resourceLimitCpu: '2000m',
             resourceRequestMemory: '1000Mi',
             resourceLimitMemory: '2000Mi',
+        ),
+        ontainerTemplate(
```
Comment (supereagle) [inline Jenkinsfile]: typo?

File: Jenkinsfile [hunk 2]
```diff
@@ -30,12 +30,28 @@ podTemplate(
...
+            name: 'docker',
+            image: 'docker:stable',
+            ttyEnabled: true,
+            command: 'cat',
```
Comment (supereagle) [inline Jenkinsfile]: Add resource request and limit?

File: Jenkinsfile [hunk 3]
```diff
@@ -52,6 +68,18 @@ podTemplate(
                 // sh('cd /go/src/github.com/caicloud/event_exporter && make push')
             }
         }
+        stage('Docker Build and push') {
+            container('docker') {
+                sh("cp /mnt/docker-hub/.dockercfg ~/.dockercfg")
+                sh("docker build -t ${dockerImage}:${shortSHA} -f Dockerfile .")
```
Comment (supereagle) [inline Jenkinsfile]: Define a variable for `${dockerImage}:${shortSHA}`?

File: Jenkinsfile (hunk 2 above)
Comment (anoopwebs) [reply Jenkinsfile]: do you have any preference? I'm not sure as I don't have any data points here.

File: Jenkinsfile (hunk 1 above)
Comment (emillg) [inline Jenkinsfile]: @anoopwebs "ontainerTemplate" should be "containerTemplate"?

**Comment by anoopwebs** [comment]
@supereagle what we can do to make it happen?

File: Jenkinsfile (hunk 2 above)
Comment (supereagle) [reply Jenkinsfile]: Can we collect some datas from Prometheus? If you don't have time for this, I think we can move on. It does not block this PR.

**Comment by supereagle** [comment]
> @supereagle what we can do to make it happen?

Missing the notification, sorry for the late reply.

File: Jenkinsfile (hunk 2 above)
Comment (supereagle) [reply Jenkinsfile]: Maybe this comment is missed, so I will move on.

> Can we collect some datas from Prometheus? If you don't have time for this, I think we can move on. It does not block this PR.

**Pull request merged**